homeassistant/components/heatzy/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""Heatzy platform configuration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from heatzypy import HeatzyClient
+from heatzypy.exception import AuthenticationFailed, HeatzyException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import API_TIMEOUT, DEBOUNCE_COOLDOWN, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = 60
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Heatzy as config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    coordinator = HeatzyDataUpdateCoordinator(hass, entry)
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
+
+class HeatzyDataUpdateCoordinator(DataUpdateCoordinator):
+    """Define an object to fetch datas."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Class to manage fetching Heatzy data API."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=SCAN_INTERVAL),
+            request_refresh_debouncer=Debouncer(
+                hass, _LOGGER, cooldown=DEBOUNCE_COOLDOWN, immediate=False
+            ),
+        )
+        self.api = HeatzyClient(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            async_create_clientsession(hass),
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Update data."""
+        try:
+            async with async_timeout.timeout(API_TIMEOUT):
+                return await self.api.async_get_devices()
+        except AuthenticationFailed as error:
+            raise ConfigEntryAuthFailed from error
```

Comment:
Should this be
```suggestion
            raise ConfigEntryAuthFailed() from error
```
or similar..?